larq_zoo/densenet.py

```diff
@@ -49,11 +50,12 @@ def binary_densenet(
 
     for block, layers_per_block in enumerate(args.layers):
         for _ in range(layers_per_block):
-            x = densely_connected_block(x, args)
+            x = densely_connected_block(x, args, args.dilation_rate[block])
 
         if block < len(args.layers) - 1:
             x = keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
-            x = keras.layers.MaxPooling2D(2, strides=2)(x)
+            if args.dilation_rate[block + 1] == 1:
```

Comment:
Should we consider the case where some but not all of the max pooling layers are replaced with dilation? (e.g. dilation_rate = [1, 1, 2, 2])

Reply:
I tried to keep this close to the [official implementation](https://github.com/hpi-xnor/BMXNet-v2-examples/blob/master/binary_models/densenet.py#L84-L87).

I think for now it is fine to only use dilation when we don't reduce the spacial dimension in the transition block.